alpha/declcfg/load.go

```diff
@@ -297,6 +293,15 @@ func LoadReader(r io.Reader) (*DeclarativeConfig, error) {
 				return fmt.Errorf("parse bundle: %v", err)
 			}
 			cfg.Bundles = append(cfg.Bundles, b)
+		case SchemaDeprecation:
+			var d Deprecation
+			if err := json.Unmarshal(in.Blob, &d); err != nil {
+				return fmt.Errorf("parse deprecation: %v", err)
+			}
+			if _, ok := cfg.Deprecations[d.Package]; ok {
+				return fmt.Errorf("deprecation for package %q already exists", d.Package)
```

Comment:
Packages, channels and bundles above don't seem to have this check. Why are deprecations different?

Reply:
Each of those types can have an arbitrary quantity but Deprecations are singletons per package.  One of the complexities here is keeping these closely-coupled objects disjoint.

Reply:
I think it's true that some set of properties on a package, channel and bundle would be uniquely identifiable, yes? Is it allowed for a catalog to contain two bundles that are identical? If yes, why? If no, why is the validation for that not here? Can the validation for deprecations live where the validation for bundles does?

Reply:
Can-o-worms territory here. Historically the sole validation of declcfg here was "can we unmarshal it".  
For `opm validate` command, it's /slightly/ more, as [can we unmarshal and create a model.Model from it](https://github.com/operator-framework/operator-registry/blob/master/pkg/lib/config/validate.go). 

I wanted a loud/early/lightweight failure if we violated singleton-per-package.

Reply:
There's no reason we couldn't implement a validation phase after completely processing the FS, but it doesn't currently exist and I didn't feel that this work rose to the level of demanding it.

Reply:
Perhaps sketch out the warpath in the future for the other parts here so it's clear why we have one validation for one segment but not the rest?

Reply:
Sorry, I thought that I understood the validation approach here, but it's only validated when it's converted to a Model type in the [Validate](https://github.com/grokspawn/operator-registry/blob/3432b12a869e37e0456567b0c93edb679f372a58/alpha/model/model.go#L49-L8) method.

I haven't gotten that far with this work yet, but that is also where we'll move this check.